docs/content/guides/dagster/pythonic-resources.mdx

```diff
@@ -0,0 +1,337 @@
+
+<TabGroup persistentKey="assetsorops">
+<TabItem name="Using software-defined-assets">
+
+#### Using software-defined assets
+
+Here, we define a <PyObject object="ConfigurableResource"/> representing a connection to an external service. We can configure the resource by constructing it in the <PyObject object="Definitions" /> call.
+
+We can also define methods on the resource which rely on the config values.
+
+```python file=/guides/dagster/pythonic_resources/pythonic_resources.py startafter=start_new_resources_configurable_defs endbefore=end_new_resources_configurable_defs
+from dagster import asset, Definitions
+from dagster._config.structured_config import ConfigurableResource
+
+
+class MyConnectionResource(ConfigurableResource):
+    username: str
+    password: str
+
+    def request(self, endpoint: str) -> Response:
+        return requests.get(
+            f"https://my-api.com/{endpoint}",
+            auth=(self.username, self.password),
+        )
+
+
+# Since MyConnectionResource extends ConfigurableResource, we don't
+# need to wrap it in a Resource[] annotation.
+@asset
+def data_from_service(my_conn: MyConnectionResource) -> Dict[str, Any]:
+    return my_conn.request("/fetch_data").json()
+
+
+defs = Definitions(
+    assets=[data_from_service],
+    resources={
+        "my_conn": MyConnectionResource(username="my_user", password="my_password"),
+    },
+)
+```
+
+</TabItem>
+<TabItem name="Using ops and jobs">
+
+#### Using ops
+
+Here, we define a <PyObject object="ConfigurableResource"/> representing a connection to an external service. We can configure the resource by constructing it in the <PyObject object="Definitions" /> call.
+
+We can also define methods on the resource which rely on the config values.
+
+```python file=/guides/dagster/pythonic_resources/pythonic_resources.py startafter=start_new_resources_configurable_defs_ops endbefore=end_new_resources_configurable_defs_ops
+from dagster import Definitions, job, op
+from dagster._config.structured_config import ConfigurableResource
+
+
+class MyConnectionResource(ConfigurableResource):
+    username: str
+    password: str
+
+    def request(self, endpoint: str) -> Response:
+        return requests.get(
+            f"https://my-api.com/{endpoint}",
+            auth=(self.username, self.password),
+        )
+
+
+# Since MyConnectionResource extends ConfigurableResource, we don't
+# need to wrap it in a Resource[] annotation.
+@op
+def update_service(my_conn: MyConnectionResource):
+    my_conn.request("/update")
+
+
+@job
+def update_service_job():
+    update_service()
+
+
+defs = Definitions(
+    jobs=[update_service_job],
+    resources={
+        "my_conn": MyConnectionResource(username="my_user", password="my_password"),
+    },
+)
+```
+
+</TabItem>
+</TabGroup>
+
+---
+
+## Using environment variables with resources
```

Comment:
Note somewhere that we have special features in Cloud to support in-UI env variables.